tests/author.test.js

```diff
@@ -0,0 +1,132 @@
+// tests/reader.test.js
+const { expect } = require("chai");
+const request = require("supertest");
+const { Author } = require("../src/models");
+const app = require("../src/app");
+
+describe("/author", () => {
+  before(async () => Author.sequelize.sync());
+
+  beforeEach(async () => {
+    await Author.destroy({ where: {} });
+  });
+
+  describe("with no records in the database", () => {
+    describe("POST /author", () => {
+      it("creates a new author in the database", async () => {
+        const response = await request(app).post("/author").send({
+          author: "Danny Wallace",
+        });
+        const newAuthorRecord = await Author.findByPk(response.body.id, {
+          raw: true,
+        });
+
+        expect(response.status).to.equal(201);
+        expect(newAuthorRecord.author).to.equal("Danny Wallace");
+      });
+
+      it("checks validation when creating new author with empty string", async () => {
+        const response = await request(app).post("/author").send({
+          author: "",
+        });
+        const newAuthorRecord = await Author.findByPk(response.body.id, {
+          raw: true,
+        });
+
+        expect(response.status).to.equal(400);
+        expect(newAuthorRecord).to.equal(null);
+      });
+    });
+  });
+
+  describe("with records in the database", () => {
+    let authors;
+
+    beforeEach(async () => {
+      authors = await Promise.all([
+        Author.create({
+          author: "J K Rowling",
+        }),
+        Author.create({
+          author: "Robert Galbraith",
+        }),
+        Author.create({
+          author: "Richard Osman",
+        }),
+      ]);
+    });
+
+    describe("GET /authors", () => {
+      it("gets all authors records", async () => {
+        const response = await request(app).get("/author");
+
+        expect(response.status).to.equal(200);
+        expect(response.body.length).to.equal(3);
+
+        response.body.forEach((author) => {
+          const expected = authors.find((a) => a.id === author.id);
+
+          expect(author.author).to.equal(expected.author);
+        });
```

Comment:
Extra brackets can be removed here, also this could be refactored to use more readable names, something like:
```suggestion
        response.body.forEach(responseAuthor => {
          const expected = authors.find(testAuthor => testAuthor.id === responseAuthor.id);

          expect(responseAuthor.author).to.equal(expected.author);
        });
```